### ISSUE TITLE
A question about your codes for fine-tune transformer

### ISSUE THREAD

**Issue by skyrover23** [andersonh-uta/Lak2023#1]
Hello professor. Thanks a lot for your codes about fine-tune transformer. I tried to applied it on my own dataset to do a two-class (0, 1) classification. The codes ran well without error warning, and the loss looks fine. However, as I check the prediction for the test sets, the outputs are all 0. I have tried to make some modifications like changing batch size but the problem has not been solved. In most cases, it will give an output of all 0. In some cases, there are very few 1 but the performance is still very bad. The problem exists for both .py and .ipynb versions. I am wondering do you have any idea about why this happened?

I am really looking forward to your reply. Thank you in advance.

**Comment by andersonh-uta** [comment]
Are you able to run the code as-is, with the Amazon sentiment data?  Is it just your data where there are problems showing up?

Without having a copy of your data or how you've modified the code, it's hard to say why your issue might be happening.  But without any further information, here's what I'd guess:

- If you have _really_ imbalanced classes, you'll need to make some adjustments to your code, otherwise the models might learn that they can minimize their loss by always predicting the majority class.
- The class/target labels need to be one-hot encoded into an NxM array, where N is the number of observations, and M is the number of possible classes.  If your labels are in a one-dimensional tensor, your results will basically just be noise.  My code is somewhat custom-tailored to the Amazon reviews data, which has 5 classes, so you might want to make sure you've got the right number of class labels for your dataset.

Most of my code was adapted from the Huggingface tutorials (https://huggingface.co/docs/transformers/training); you might want to give that a read to make sure there isn't some code-related thing you've overlooked.

**Comment by skyrover23** [comment]
Hi professor. Thanks a lot for your response. I will try to find the mistakes in my codes according to your suggestions.
